framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/AClassWrapper.java

```diff
@@ -0,0 +1,169 @@
+     * <p>Results are interened.
+     *
+     * @param methodName the name of the method
+     * @return an AMethodWrapper representing the method
+     */
+    public AMethodWrapper vivifyMethod(String methodName) {
+        if (methods.containsKey(methodName)) {
+            return methods.get(methodName);
+        } else {
+            AMethodWrapper wrapper = new AMethodWrapper(theClass.methods.getVivify(methodName));
+            methods.put(methodName, wrapper);
+            return wrapper;
+        }
+    }
+
+    /**
+     * Get all the methods that have been vivified (that is, interacted with) on a class.
+     *
+     * @return a map from method name to the object representing the method
+     */
+    public Map<String, AMethodWrapper> getMethods() {
+        return ImmutableMap.copyOf(methods);
+    }
+
+    /**
+     * Call before doing anything with a field. Interacts with scenelib and fetches or creates an
+     * appropriate AFieldWrapper object.
+     *
+     * <p>Results are interned.
+     *
+     * @param fieldName the name of the field
+     * @param type the type of the field, which scenelib doesn't track
+     * @return an AField object representing the field
+     */
+    public AFieldWrapper vivifyField(String fieldName, TypeMirror type) {
+        if (fields.containsKey(fieldName)) {
+            return fields.get(fieldName);
+        } else {
+            AFieldWrapper wrapper =
+                    new AFieldWrapper(theClass.fields.getVivify(fieldName), type.toString());
+            fields.put(fieldName, wrapper);
+            return wrapper;
+        }
+    }
+
+    /**
+     * Get all the fields that have been vivified (that is, interacted with) on a class.
+     *
+     * @return a map from field name to the object representing the field
+     */
+    public Map<String, AFieldWrapper> getFields() {
+        return ImmutableMap.copyOf(fields);
+    }
+
+    /**
+     * Get the annotations on the class
+     *
+     * @return the annotations, directly from scenelib
+     */
+    public Collection<? extends Annotation> getAnnotations() {
+        return theClass.tlAnnotationsHere;
+    }
+
+    /**
+     * The type of the class
+     *
+     * @return a type element representing this class
+     */
+    public TypeElement getTypeElement() {
+        return typeElement;
+    }
+
+    /**
+     * Set the type element representing the class
+     *
+     * @param typeElement the type element representing the class
+     */
+    public void setTypeElement(TypeElement typeElement) {
+        if (this.typeElement == null) {
+            this.typeElement = typeElement;
+        }
+    }
+
+    /**
+     * Checks if any enum constants have been provided to this class
+     *
+     * @return true if this class has been marked as an enum
+     */
+    public boolean isEnum() {
+        return enumConstants != null;
+    }
+
+    /**
+     * Returns the set of enum constants for this class, or null if this is not an enum
```

Comment:
I don't see how this can return null.  It would throw a NullPointerException instead.